opa-filter-autoconfigure/src/test/groovy/com/bisnode/opa/spring/security/filter/autoconfigure/OpaFilterConfigurerSpec.groovy

```diff
@@ -144,13 +144,30 @@ class OpaFilterConfigurerSpec extends Specification {
           e.response.status == 401
     }
 
+    def 'should not ask OPA for whitelisted path and authenticated user'() {
+        given:
+          wireMockServer.stubFor(any(anyUrl())
+                  .willReturn(aResponse()
+                          .withStatus(200)
+                          .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)
+                          .withBody('{"result": {"allow": false}}')
```

Comment:
This makes it look like OPA is handling the response when really it isn't.

Reply:
I wasn't sure how to test to "don't ask OPA" so I simulated that OPA returns deny and nonetheless I get http status 200 and no exceptions thrown. Do you have any idea how to test it in different way?

Reply:
I'd go with interaction-based testing. First, you can check, that mocked OpaQueryApi is not called, second you can use wiremock verifier to verify that no interactions were made.

Reply:
I used wiremock to verify that no interactions with path "/whitelisted" were made.